docs/release-process.md

```diff
@@ -0,0 +1,129 @@
+You do not need to build the nightly package variants or other OS bases as
+these are not part of the released images. You can push the images to quay.io
+under your own repos to perform a early security scan as well.
+
+If you are happy with the content of the `release` branch, tag it. Example:
+
+```
+git checkout release
+git tag -a -m 'Release v0.3' v0.3
+```
+
+This creates an annotated tag. Release tags must be annotated tags.
+
+### Build
+
+Using the tagged `release` branch, the container images for release will be
+built. It is very important to ensure that base images are up-to-date.
+It is very important to ensure that you perform the next set of steps with
+clean new builds and do not use cached images. To accomplish both tasks it
+is recommended to purge your local container engine of cached images
+(Example: `podman image rm --all`). You should have no images named like
+`quay.io/samba.org` in your local cache.
+
+Build the images from scratch. Example:
+```
+make build-ad-server build-server build-client build-toolbox
+```
+
+For each image that was just built, apply a temporary pre-release tag
+to it. Example:
+```
+for img_name in ad-server server client toolbox ; do
+    podman tag quay.io/samba.org/samba-${img_name}:{latest,v0.3pre1}
+done
+```
+
+Log into quay.io.  Push the images to quay.io using the temporary tag. Example:
+```
+for img_name in ad-server server client toolbox ; do
+    podman push quay.io/samba.org/samba-${img_name}:v0.3pre1
+done
+```
+
+Wait for the security scan to complete. There shouldn't be any issues if you
+properly updated the base images before building. If there are issues and you
+are sure you used the newest base images, check the base images on quay.io and
+make sure that the number of issues are identical. The security scan can take
+some time, while it runs you may want to do other things.
+
+
+## GitHub Release
+
+When you are satisfied that the tagged version is suitable for release, you
+can push the tag to the public repo:
+```
+git push --follow-tags
+```
+
+Draft a new set of release notes. Select the recently pushed tag. Start with
+the auto-generated release notes from GitHub (activate the `Generate release
+notes` button/link). Add an introductory section (see previous notes for an
+example). Add a "Highlights" section if there are any notable features or fixes
+in the release. The Highlights section can be skipped if the content of the
+release is unremarkable (e.g. few changes occurred since the previous release).
+
+Because this is a container based release we do not provide any build artifacts
+on GitHub (beyond the sources automatically provided there). Instead we add
+a Downloads section that notes the exact tags and digests that the images can
+be found at on quay.io.
+
+Use the following partial snippet as an example:
+```
+Images built for this release can be acquired from the quay.io image registry.
```

Comment:
```suggestion
Images built for this release can be obtained  from the quay.io image registry.
```

Reply:
The words are pretty much synonyms in my head, but sure. Done.

Reply:
> The words are pretty much synonyms in my head, but sure. Done.

Thanks! I understood the intention originally but couldn't resist because "acquire" always sounds to me like costs are involved ;-)